cassette/http_connection.py

```diff
@@ -124,6 +132,12 @@ class UL3CassetteHTTPConnection(CassetteConnectionMixin,
                                     requests_urllib3.connection.HTTPConnection):
 
         _baseclass = requests_urllib3.connection.HTTPConnection
+        # requests 2.3.0 and below have an issue where they get confused if
+        # the HTTPConnection has a "sock" attribute which is set to None at
+        # the end of a request/response cycle. So we delete the attribute in
+        # those cases
+        _delete_sock_when_returning_from_library = True if \
```

Comment:
Why the ternary and not just `semver.compare(requests.__version__, '2.4.0') == -1`?